src/power_grid_model/data_types.py

```diff
@@ -7,84 +7,210 @@
 have been defined and explained in this file.
 """
 
-from typing import TypeVar, Union
+from typing import TypeAlias, TypeVar, Union
 
 import numpy as np
 
-from power_grid_model.core.dataset_definitions import ComponentTypeLike  # pylint: disable=unused-import
 from power_grid_model.core.dataset_definitions import ComponentTypeVar
 
+AttributeType: TypeAlias = str
+"""
+An attribute type is a string reprenting the attribute type of a specific component.
+
+- Examples:
+
+    - "id"
+    - "u_rated"
+"""
+
+SparseDataComponentType: TypeAlias = str
+"""
+A string representing the component type of sparse data structures.
+
+Must be either "data" or "indptr".
+"""
+
 SingleArray = Union[np.ndarray]
 """
-A single array is a one-dimensional structured containing a list of components of the same type.
+A single array is a one-dimensional structured numpy array containing a list of components of the same type.
 
 - Examples:
 
     - structure: <1d-array>
     - concrete: array([(0, 10500.0), (0, 10500.0)], dtype=power_grid_meta_data["input"]["node"].dtype)
 """
 
+SingleColumn = Union[np.ndarray]
+"""
+A single column is a one-dimensional structured numpy array containing a list of the same attribute of
+multiple components of the same type.
+
+- Examples:
+
+    - structure: <1d-array>
+    - concrete:
+
```

Comment:
Remove line.

Reply:
no because this is how Sphinx works to generate the documentation in readthedocs